src/context.ts

```diff
@@ -993,6 +993,35 @@ export class Context implements RenamedUpdate {
         );
     }
 
+    /**
+     * Context-aware alias for `api.sendMessageDraft`. Use this method to stream a partial message to a user while the message is being generated; supported only for bots with forum topic mode enabled. Returns True on success.
+     *
+     * @param text Text of the message to be sent, 1-4096 characters after entities parsing
+     * @param other Optional remaining parameters, confer the official reference below
+     * @param signal Optional `AbortSignal` to cancel the request
+     *
+     * **Official reference:** https://core.telegram.org/bots/api#sendmessagedraft
+     */
+    replyWithDraft(
+        text: string,
+        other?: Other<"sendMessageDraft", "chat_id" | "text">,
+        signal?: AbortSignal,
+    ) {
+        const msg = this.msg;
+        return this.api.sendMessageDraft(
+            orThrow(this.chatId, "sendMessageDraft"),
+            this.update.update_id,
+            text,
+            {
+                ...(msg?.is_topic_message
```

Comment:
question (non-blocking): will this ever be false? and, if it is, will the API call fail? should we error preemptively to avoid an unnecessary API call and provide a clear error?

Reply:
Yes, it is false if the bot does not have topics enabled. grammY should then call the method without the thead identifier as usual.

The API call will fail with the error message that the Bot API server generates. For example, if it ever decides to support this message without a thread identifier, then this code will keep on performing the right thing.